idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -1536,3 +1535,199 @@ def sample_points(self):
+                - When the aim is to generate a set of samples from a data range, the dataset must be a list containing two lists of equal lengths which contain the variable bounds and **sampling_type** option must be set to "creation". It is assumed that the range contains no output variable information in this case.
+
+            number_of_samples(int): The number of samples to be generated. Should be a positive integer less than or equal to the number of entries (rows) in **data_input**.
+            list_of_distributions (list): The list containing the probability distribution for each variable. The length of the list must match the number of input (i.e. dependent) variables to be sampled. We currently support random, uniform and normal (i.e. Gaussian) distributions.
+            sampling_type (str) : Option which determines whether the algorithm selects samples from an existing dataset ("selection") or attempts to generate sample from a supplied range ("creation"). Default is "creation".
+
+        Keyword Args:
+            xlabels (list): List of column names (if **data_input** is a dataframe) or column numbers (if **data_input** is an array) for the independent/input  variables.  Only used in "selection" mode. Default is None.
+            ylabels (list): List of column names (if **data_input** is a dataframe) or column numbers (if **data_input** is an array) for the dependent/output variables. Only used in "selection" mode. Default is None.
+
+        Returns:
+            **self** function containing the input information
+
+        Raises:
+            ValueError: The **data_input** is the wrong type
+
+            ValueError: When a non-implemented distribution is supplied in list_of_distributions
+
+            IndexError: When invalid column names are supplied in **xlabels** or **ylabels**
+
+            Exception: When the **number_of_samples** is invalid (not an integer, too large, zero, negative)
+
+
+        """
+        if sampling_type is None:
+            sampling_type = "creation"
+            self.sampling_type = sampling_type
+            print("Creation-type sampling will be used.")
+        elif not isinstance(sampling_type, str):
+            raise Exception("Invalid sampling type entry. Must be of type <str>.")
+        elif (sampling_type.lower() == "creation") or (
+            sampling_type.lower() == "selection"
+        ):
+            sampling_type = sampling_type.lower()
+            self.sampling_type = sampling_type
+        else:
+            raise Exception(
+                'Invalid sampling type requirement entered. Enter "creation" for sampling from a range or "selection" for selecting samples from a dataset.'
+            )
+        print("Sampling type: ", self.sampling_type, "\n")
```

Comment:
This is probably a comment for the future, but we should probably switch to using loggers for all this output so that we can control how much gets seen and where.

Reply:
Yes, we can always change that later.